apps/api/src/services/handles.service.ts

```diff
@@ -1,18 +1,57 @@
-import { Injectable } from '@nestjs/common';
+import { Injectable, Logger } from '@nestjs/common';
 import { InjectRedis } from '@liaoliaots/nestjs-redis';
 import Redis from 'ioredis';
 import { InjectQueue } from '@nestjs/bullmq';
-import { QueueConstants } from '../../../../libs/common/src';
+import { AccountChangeRepsonseDto, QueueConstants } from '../../../../libs/common/src';
 import { BlockchainService } from '../../../../libs/common/src/blockchain/blockchain.service';
-import { AccountResponse } from '../../../../libs/common/src/dtos/accounts.dto';
-import { HandlesResponse } from '../../../../libs/common/src/dtos/handles.dtos';
+import { HandlesRequest, HandlesResponse } from '../../../../libs/common/src/dtos/handles.dtos';
+import { Queue } from 'bullmq';
+import { ConfigService } from '../../../../libs/common/src/config/config.service';
+import type { HandleResponse, MessageSourceId } from '@frequency-chain/api-augment/interfaces';
+import { createHash } from 'crypto';
+import { AccountChangeType } from '../../../../libs/common/src/dtos/account.change.notification.dto';
 
 @Injectable()
 export class HandlesService {
-  constructor(private blockchainService: BlockchainService) {}
+  private readonly logger: Logger;
 
-  createHandle(handle: string): string {
-    return 'Handle created successfully: ' + handle;
+  constructor(
+    @InjectRedis() private redis: Redis,
+    @InjectQueue(QueueConstants.ACCOUNT_CHANGE_PUBLISH_QUEUE)
+    private accountChangePublishQueue: Queue,
+    private configService: ConfigService,
+    private blockchainService: BlockchainService,
+  ) {
+    this.logger = new Logger(this.constructor.name);
+  }
+
+  private calculateJobId(jobWithoutId): string {
+    const stringVal = JSON.stringify(jobWithoutId);
+    return createHash('sha1').update(stringVal).digest('base64url');
+  }
+
+  async enqueueRequest(
+    request: HandlesRequest,
+    type: AccountChangeType,
+  ): Promise<AccountChangeRepsonseDto> {
+    const providerId = this.configService.getProviderId();
+    const data = {
+      ...request,
+      type,
+      providerId,
+      referenceId: this.calculateJobId(request),
```

Comment:
Question: Do we need a hash/referenceId? We might be able to skip `enqueueRequest` altogether and just go to `this.accountChangePublishQueue.add()`. 🤔 
Just using the `job` returned from `.add()` might be enough for our purposes.

Reply:
My thinking is that then we would have a new service for each instance we add, rather than just one - enqueueRequest. We could also start accessing the blockchain service directly from the controller, but I was under the impression that was kinda bad form. Thoughts?